src/components/layer-controller/ChannelController.js

```diff
@@ -15,6 +15,14 @@ export const toRgbUIString = (on, arr, theme) => {
   return `rgb(${color})`;
 };
 
+function truncateDecimalNumber(value, maxLength) {
+  if (!value && value !== 0) return '';
+  const stringValue = value.toString();
+  return stringValue.length > maxLength
+    ? stringValue.substring(0, maxLength).replace(/\.$/, '')
```

Comment:
This doesn't seem right: would number.toString ever end with `.`?

Reply:
The substring could since "6553.121" would be truncated to "6553.".  Does that make sense as an edge case?

Reply:
I can add a comment to make that clearer.

Reply:
If you want to test this out with some real data (console.log'ing to check), you can replace line 68 of `api.js` with this, inserting you hubmap token in the two spots:

```javascript
const vanderbiltBase = {
  description: vanderbiltDescription,
  layers: [
    {
      name: 'raster',
      type: 'RASTER',
      fileType: 'raster.json',
      url: URL.createObjectURL(
        new Blob([
          JSON.stringify({
            schemaVersion: '0.0.2',
            images: [
              {
                name: 'IMC 32 bit',
                url:
                  'https://assets.hubmapconsortium.org/b5e993ba9f01b0f5b8d1b6f4fbabc60a/ometiff-pyramids/ometiff/20191202_HuBMAP_Thymus/20191202_HuBMAP_Thymus_s0_p8_r3_a3_ac.ome.tif?token=my_token',
                type: 'ome-tiff',
                metadata: {
                  omeTiffOffsetsUrl:
                    'https://assets.hubmapconsortium.org/b5e993ba9f01b0f5b8d1b6f4fbabc60a/output_offsets/ometiff/20191202_HuBMAP_Thymus/20191202_HuBMAP_Thymus_s0_p8_r3_a3_ac.offsets.json?token=my_token',
                },
              },
            ],
          }),
        ]),
      ),
    },
  ],
};
```

Then go to the "Spraggins" dataset from the homepage.  Sorry, I should have included this.

Reply:
I guess you know that you're not going to specify a number so large or a length so short that digits don't get chopped off the end, but I don't think we want to require future devs to figure that out.

Would plain `Math.round()` work? Or if you want to keep the decimals in smaller numbers, how about
```
Intl.NumberFormat('en-US', { maximumSignificantDigits: 4 , useGrouping: false }).format(6553.121);
"6553"
Intl.NumberFormat('en-US', { maximumSignificantDigits: 4 , useGrouping: false }).format(6.553121);
"6.553"
Intl.NumberFormat('en-US', { maximumSignificantDigits: 4 , useGrouping: false }).format(6553121);
"6553000"
```

Reply:
I had no idea that existed!  Let's talk about this tomorrow in stand-up.  I think this is a good solution in terms of readability, but I worry about large numbers.  My solution is not better though so perhaps replacing large numbers with something like `4.23e6` would be better and also satisfy the `maxLength` constraint I want to enforce.